### DISCUSSION TITLE
Is there a way to hide/disable the plugin on the fly.

### DISCUSSION THREAD

**Discussion by bassamsdata** [Bekaboo/dropbar.nvim#204] [Q&A] [answered]
I don’t want to use the plugin all the time, sometimes it just distracts me.  

Is there an API I can use to hide/disable the plugin and then enable it later (like a toggle command)?  

Thank you!

**Comment by Bekaboo** [comment, marked as answer]
You can try to customize your `bar.enable` function so that it won't attach to current window if window-local variable `winbar_no_attach` is set, like this:

```lua
require('dropbar').setup({
  bar = {
    enable = function(buf, win, _)
      return not vim.w[win].winbar_no_attach -- add this line
        and vim.api.nvim_buf_is_valid(buf)
        and vim.api.nvim_win_is_valid(win)
        and vim.wo[win].winbar == ''
        and vim.fn.win_gettype(win) == ''
        and vim.bo[buf].ft ~= 'help'
        and ((pcall(vim.treesitter.get_parser, buf)) and true or false)
    end,
  },
})
```

Then to disable dropbar in current window, you can just use the command `:setlocal winbar= | let w:winbar_no_attach = v:true`. To re-enable dropbar in current window, you can use the command `:unlet w:winbar_no_attach | setlocal winbar='%{%v:lua.dropbar.get_dropbar_str()%}'`

**Reply by bassamsdata** [reply]
thank you so much for the detailed reply. it works like a charm.

I did a keymap for it with lua syntax for anyone interested.
```lua
local function toggle_dropbar()
  local win = vim.api.nvim_get_current_win()
  if vim.w[win].winbar_no_attach then
    -- Re-enable dropbar
    vim.w[win].winbar_no_attach = false
    vim.wo[win].winbar = "%{%v:lua.dropbar.get_dropbar_str()%}"
    vim.notify("Dropbar enabled", vim.log.levels.INFO)
  else
    -- Disable dropbar
    vim.wo[win].winbar = ""
    vim.w[win].winbar_no_attach = true
    vim.notify("Dropbar disabled", vim.log.levels.INFO)
  end
end


vim.keymap.set("n", "<leader>ur", toggle_dropbar, { desc = "Toggle dropbar" })

```
edit: typo
